### ISSUE TITLE
Add test cases for different handshake versions 

### ISSUE THREAD

**Issue by kjellwinblad** [ElMaxo/brod_gssapi#13]
There are three handshake versions undefined, 0 and 1 (see https://github.com/kafka4beam/brod_gssapi/pull/5#discussion_r894101153). @starbelly, please correct me if I did not understand this correctly. We should add Kafka test cases and github actions for all handshake versions by extending the code in the example directory.

**Comment by starbelly** [comment]
@kjellwinblad That's correct. With auth/8 function added to kafka_protocol that will invoke auth/7 on a auth plugin, we'll now get handshake `0` or `1`. We may also get undefined from ourselves if auth/6 is invoked from kafka_protocol and we set it ourselves (to undefined). 

However, we currently don't support handshake v0. See issue #8.
